### ISSUE TITLE
problem with matching function when using default arguments

### ISSUE THREAD

**Issue by ferdymercury** [doxygen/doxygen#9144]
**Describe the bug**
Doxygen thinks there are several candidates to a function when there is only one, probably because of the fact that a "using namespace" statement is contained in the .cxx file.

**Expected behavior**
No warning is raised.

**Screenshots**
```
/opt/root_src/graf3d/eve7/src/REveDigitSet.cxx:299: warning: no uniquely matching class member found for 
  void REveDigitSet::DigitColor(UChar_t r, UChar_t g, UChar_t b, UChar_t a)
Possible candidates:
  'void ROOT::Experimental::REveDigitSet::DigitColor(Color_t ci)' at line 116 of file /opt/root_src/graf3d/eve7/inc/ROOT/REveDigitSet.hxx
  'void ROOT::Experimental::REveDigitSet::DigitColor(Color_t ci, Char_t transparency)' at line 117 of file /opt/root_src/graf3d/eve7/inc/ROOT/REveDigitSet.hxx
  'void ROOT::Experimental::REveDigitSet::DigitColor(UChar_t r, UChar_t g, UChar_t b, UChar_t a=255)' at line 118 of file /opt/root_src/graf3d/eve7/inc/ROOT/REveDigitSet.hxx
  'void ROOT::Experimental::REveDigitSet::DigitColor(UChar_t *rgba)' at line 119 of file /opt/root_src/graf3d/eve7/inc/ROOT/REveDigitSet.hxx
```

**To Reproduce**
Use reproducer at https://github.com/doxygen/doxygen/issues/9143

**Version**
1.9.4 (15dae504a167d838873053cc96417d8beae78df7*)

**Comment by albert-github** [comment]
Problem looks like to be coming from the `UChar_t a=255` part, when using `UChar_t a` no warning is emitted.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9146

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
